### DISCUSSION TITLE
Is there a way to open a termial inside lazygit?

### DISCUSSION THREAD

**Discussion by NilCent** [jesseduffield/lazygit#4071] [Q&A]
I want to search for a commit message. like ` git log --grep="tidrange"`

To do that, i have to open tmux first.

Is there a way to open a terminal using just lazygit?

By the way, Is there a way to search for commit message by lazygit?

**Comment by mark2185** [comment]
If you want to add custom commands you use more often, you can add them to your config as shown [here](https://github.com/jesseduffield/lazygit/blob/master/docs/Custom_Command_Keybindings.md).

For running a one-off shell command, you can just run it through the menu under `:`.

> By the way, Is there a way to search for commit message by lazygit?

By going to the commits panel (`3`), you can search by pressing `/`.

All of these things are in the help menu that is opened by pressing `?` in any given panel.

**Reply by stefanhaller** [reply]
> > By the way, Is there a way to search for commit message by lazygit?
> 
> By going to the commits panel (`3`), you can search by pressing `/`.

However, this only searches in the commit message subjects. There is currently no way to search in the message bodies.

**Reply by NilCent** [reply]
Thanks, but I’m seeking a method to launch a terminal with lazygit in the same display, comparable to the `:term` feature in vim, or to use tmux for this purpose.
